pkg/agent/datapath/multiBridgeDatapath.go

```diff
@@ -374,6 +287,124 @@ func (datapathManager *DpManager) InitializeDatapath(stopChan <-chan struct{}) {
 	}()
 }
 
+func NewVDSForConfig(datapathManager *DpManager, vdsID, ovsbrname string) {
+	// initialize vds bridge chain
+	localBridge := NewLocalBridge(ovsbrname, datapathManager)
+	policyBridge := NewPolicyBridge(ovsbrname, datapathManager)
+	clsBridge := NewClsBridge(ovsbrname, datapathManager)
+	uplinkBridge := NewUplinkBridge(ovsbrname, datapathManager)
+	vdsBridgeMap := make(map[string]Bridge)
+	vdsBridgeMap[LOCAL_BRIDGE_KEYWORD] = localBridge
+	vdsBridgeMap[POLICY_BRIDGE_KEYWORD] = policyBridge
+	vdsBridgeMap[CLS_BRIDGE_KEYWORD] = clsBridge
+	vdsBridgeMap[UPLINK_BRIDGE_KEYWORD] = uplinkBridge
+
+	// initialize of controller
+	vdsOfControllerMap := make(map[string]*ofctrl.Controller)
+	vdsOfControllerMap[LOCAL_BRIDGE_KEYWORD] = ofctrl.NewController(localBridge)
+	vdsOfControllerMap[POLICY_BRIDGE_KEYWORD] = ofctrl.NewController(policyBridge)
+	vdsOfControllerMap[CLS_BRIDGE_KEYWORD] = ofctrl.NewController(clsBridge)
+	vdsOfControllerMap[UPLINK_BRIDGE_KEYWORD] = ofctrl.NewController(uplinkBridge)
+
+	go vdsOfControllerMap[LOCAL_BRIDGE_KEYWORD].Listen("0")
+	go vdsOfControllerMap[POLICY_BRIDGE_KEYWORD].Listen("0")
+	go vdsOfControllerMap[CLS_BRIDGE_KEYWORD].Listen("0")
+	go vdsOfControllerMap[UPLINK_BRIDGE_KEYWORD].Listen("0")
+
+	// initialize ovsdbDriver
+	vdsOvsdbDriverMap := make(map[string]*ovsdbDriver.OvsDriver)
+	bridgeSuffixToNameMap := map[string]string{
+		LOCAL_BRIDGE_KEYWORD:  localBridge.name,
+		POLICY_BRIDGE_KEYWORD: policyBridge.name,
+		CLS_BRIDGE_KEYWORD:    clsBridge.name,
+		UPLINK_BRIDGE_KEYWORD: uplinkBridge.name,
+	}
+	var wg sync.WaitGroup
+	for suffix, brName := range bridgeSuffixToNameMap {
+		wg.Add(1)
+		go func(suffix, brName string, vdsOvsdbDriverMap map[string]*ovsdbDriver.OvsDriver) {
+			defer wg.Done()
+			vdsOvsdbDriverMap[suffix] = ovsdbDriver.NewOvsDriver(brName)
+		}(suffix, brName, vdsOvsdbDriverMap)
+	}
+	wg.Wait()
+
+	datapathManager.DpManagerMutex.Lock()
```

Comment:
Is the mutex lock necessary here? Each func call NewVDSForConfig has unique vds id.